mmdet/models/dense_heads/detr_head.py

```diff
@@ -306,8 +197,6 @@ def loss_by_feat(
             dict[str, Tensor]: A dictionary of loss components.
         """
         # NOTE defaultly only the outputs from the last feature scale is used.
-        all_cls_scores = all_cls_scores_list[-1]
-        all_bbox_preds = all_bbox_preds_list[-1]
         assert batch_gt_instances_ignore is None, \
             'Only supports for batch_gt_instances_ignore setting to None.'
 
```

Comment:
![image](https://user-images.githubusercontent.com/26483343/191040247-6b87a2cc-4e99-45c7-a9a2-41fc892de763.png)


The copy of `gt_instance_list` can be avoided by the following way if `batch_gt_instances` is not modified in every `loss_by_feat_single`.


```python
losses_cls, losses_bbox, losses_iou = multi_apply(
            self.loss_by_feat_single, all_cls_scores, all_bbox_preds,
            batch_gt_instances=batch_gt_instances, batch_img_metas=batch_img_metas)
```

Please check whether this optimazation can be done.